pkg/reconciler/route/reconcile_resources_test.go

```diff
@@ -257,6 +258,28 @@ func TestReconcileCertificateUpdate(t *testing.T) {
 	}
 }
 
+func TestReconcileIngressClassAnnotation(t *testing.T) {
+	ctx, _, reconciler, _, cancel := newTestReconciler(t)
+	defer cancel()
+
+	const expClass = "foo.ingress.networking.knative.dev"
+
+	r := Route("test-ns", "test-route")
+	ci := newTestIngress(t, r)
+
+	// Add ingress.class annotation.
+	ci.ObjectMeta.Annotations[networking.IngressClassAnnotationKey] = expClass
```

Comment:
Don't we need to reconcile first, then update the annotation, then verify it actually changed?

Reply:
Yes, you should do that. Thank you. Updated.